vms/platformvm/txs/fee/calculator.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fee
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/vms/platformvm/upgrade"
+)
+
+var _ txs.Visitor = (*calculator)(nil)
+
+func NewStaticCalculator(config StaticConfig, upgradeTimes upgrade.Config) Calculator {
```

Comment:
Why do we create a struct here? It seems like every time we call this function we just immediately call `CalculateFee` only once. Do we pass this `Calculator` object around in the future?

Reply:
We don't pass this struct around really. It's meant as a wrapper to simplify the API

Reply:
Hm - so would it make more sense for this to just be a function rather than a struct?

Reply:
We want to be able to build the `calculator` with relevant fields and later call `CalculateFee`.
Note tha the upcoming Dynamic fees calculator will be inited with different fields.
I think the struct is the simplest solution here, but happy to chat about this